test/e2e-test-application/cypress/e2e/tests/1-angular/luigi-client-link-manager-features-6.cy.js

```diff
@@ -0,0 +1,102 @@
+import { wrap } from "module";
+
+
+describe('Luigi Client linkManager Modal', () => {
+  beforeEach(() => {
+    cy.visitLoggedIn('/');
+  });
+
+    it('print stuff out so, I can check what functions are available etc.', () => {
+      cy.window().then(win => {
+        cy.log(win)
+        cy.log("----win.Luigi----")
+        cy.log(win.Luigi)
+        cy.log("----win.Luigi.navigation()----")
+        cy.log(win.Luigi.navigation())
+        cy.log("----win.Luigi.navigation().openAsModal() without cy.wrap----")
+        cy.log(win.Luigi.navigation().openAsModal('/projects/pr2'))
+
+        const p = win.Luigi.navigation().openAsModal('/projects/pr2')
```

Comment:
For future reference `win.Luigi.navigation()....` is something we have done in the past but its bad practice to access Luigi Object or any other JS Object on a Cypress test.

Instead you would add an HTML Element =(button or whatever) and attach an onclick method on the code (Our Angular E2E App or other) that runs the Luigi API call.
The Luigi API call itself usually affects the DOM: Ex: Opens a modal that is present in the DOM.

So in short Cypress is good for detecting Visual DOM changes, so keep that in mind when writing tests with it.

Reply:
Ah I see, good to know :)